---
title: "McH1-7 Field Trials"
author: "J. Meyer"
date: "2024-03-05"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load libraries

```{r, echo=FALSE}
library(dada2)
library(ggplot2)
library(phyloseq)
library(vegan)
library(pairwiseAdonis)
library(knitr)
library(CoDaSeq)
library(dplyr)
library(cowplot)
library(randomcoloR)
library(reshape2)
writeLines(capture.output(sessionInfo()), "sessionInfo.txt")
```

## Quality-filter the sequencing reads and create Amplicon Sequence Variant (ASV) tables with DADA2

Put unjoined R1 and R2 fastq files, with adaptors and primers previously removed with cutadapt into a directory for DADA2. Here, our forward and reverse fastq filenames have format: SAMPLENAME_R1_cut.fastq.gz and SAMPLENAME_R2_cut.fastq.gz

*****If you have samples from multiple sequencing runs, you need to determine the sequence variants for each run separately, then merge the ASV tables.
Here is the dada2 page on merging runs: https://benjjneb.github.io/dada2/bigdata_paired.html

```{r, echo=FALSE}
#### 1st sequencing run for BS2
path <- "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2107_BS2"
list.files(path)
fnFs <- sort(list.files(path, pattern="_R1_cut.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_cut.fastq.gz", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
# Perform filtering and trimming
filt_path <- file.path(path, "filtered") 
filtFs <- file.path(filt_path, paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sample.names, "_R_filt.fastq.gz"))
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(150,150),
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE)
head(out)
# Learn the Error Rates, it TAKES TIME!
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
# Dereplicate the filtered fastq files
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
names(derepFs) <- sample.names
names(derepRs) <- sample.names
# Infer the sequence variants in each sample
dadaFs <- dada(derepFs, err=errF, multithread=TRUE)
dadaRs <- dada(derepRs, err=errR, multithread=TRUE)
# Inspecting the dada-class object returned by dada:
dadaFs[[1]]
# Merge the denoised forward and reverse reads:
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
# Construct sequence table
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(mergers, getN), rowSums(seqtab))
colnames(track) <- c("input", "filtered", "denoised", "merged", "tabled")
rownames(track) <- sample.names
head(track)
write.table(track, "dada_read_stats_NS2107_BS2.txt",sep="\t",col.names=NA)
saveRDS(seqtab, "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2107_BS2/seqtab.rds") 
```


```{r, echo=FALSE}
#### 2nd sequencing run for BS2
path <- "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2172_BS2"
list.files(path)
fnFs <- sort(list.files(path, pattern="_R1_cut.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_cut.fastq.gz", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
# Perform filtering and trimming
filt_path <- file.path(path, "filtered") 
filtFs <- file.path(filt_path, paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sample.names, "_R_filt.fastq.gz"))
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(150,150),
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE)
head(out)
# Learn the Error Rates, it TAKES TIME!
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
# Dereplicate the filtered fastq files
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
names(derepFs) <- sample.names
names(derepRs) <- sample.names
# Infer the sequence variants in each sample
dadaFs <- dada(derepFs, err=errF, multithread=TRUE)
dadaRs <- dada(derepRs, err=errR, multithread=TRUE)
# Inspecting the dada-class object returned by dada:
dadaFs[[1]]
# Merge the denoised forward and reverse reads:
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
# Construct sequence table
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(mergers, getN), rowSums(seqtab))
colnames(track) <- c("input", "filtered", "denoised", "merged", "tabled")
rownames(track) <- sample.names
head(track)
write.table(track, "dada_read_stats_NS2172_BS2.txt",sep="\t",col.names=NA)
saveRDS(seqtab, "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2172_BS2/seqtab.rds")
```


```{r, echo=FALSE}
#### 3rd sequencing run for BS2
path <- "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2211_BS2"
list.files(path)
fnFs <- sort(list.files(path, pattern="_R1_cut.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_cut.fastq.gz", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
# Perform filtering and trimming
filt_path <- file.path(path, "filtered") 
filtFs <- file.path(filt_path, paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sample.names, "_R_filt.fastq.gz"))
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(150,150),
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE)
head(out)
# Learn the Error Rates, it TAKES TIME!
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
# Dereplicate the filtered fastq files
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
names(derepFs) <- sample.names
names(derepRs) <- sample.names
# Infer the sequence variants in each sample
dadaFs <- dada(derepFs, err=errF, multithread=TRUE)
dadaRs <- dada(derepRs, err=errR, multithread=TRUE)
# Inspecting the dada-class object returned by dada:
dadaFs[[1]]
# Merge the denoised forward and reverse reads:
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
# Construct sequence table
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(mergers, getN), rowSums(seqtab))
colnames(track) <- c("input", "filtered", "denoised", "merged", "tabled")
rownames(track) <- sample.names
head(track)
write.table(track, "dada_read_stats_NS2211_BS2.txt",sep="\t",col.names=NA)
saveRDS(seqtab, "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2211_BS2/seqtab.rds")
```


```{r, echo=FALSE}
#### 4th sequencing run for BS2
path <- "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2408_BS2"
list.files(path)
fnFs <- sort(list.files(path, pattern="_R1_cut.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_cut.fastq.gz", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "-V4"), `[`, 1)

# Perform filtering and trimming
filt_path <- file.path(path, "filtered") 
filtFs <- file.path(filt_path, paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sample.names, "_R_filt.fastq.gz"))
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(150,150),
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE)
head(out)
# Learn the Error Rates, it TAKES TIME!
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
# Dereplicate the filtered fastq files
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
names(derepFs) <- sample.names
names(derepRs) <- sample.names
# Infer the sequence variants in each sample
dadaFs <- dada(derepFs, err=errF, multithread=TRUE)
dadaRs <- dada(derepRs, err=errR, multithread=TRUE)
# Inspecting the dada-class object returned by dada:
dadaFs[[1]]
# Merge the denoised forward and reverse reads:
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
# Construct sequence table
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(mergers, getN), rowSums(seqtab))
colnames(track) <- c("input", "filtered", "denoised", "merged", "tabled")
rownames(track) <- sample.names
head(track)
write.table(track, "dada_read_stats_NS2408_BS2.txt",sep="\t",col.names=NA)
saveRDS(seqtab, "~/Documents/McH1-7_field_trials_BS2andBS3/cutadapt_NS2408_BS2/seqtab.rds")

##### Run NS2408 had six samples (all from October collection) with fewer than 1,000 tabled reads. Remove these samples and rerun dada2.
# samples removed: BS-2080-HD-O, BS2-2460-HD-O, BS2-2478-HD-O, BS2-2077-DD-O, BS2-2077-HD-O, BS2-2065-HH-O 
# don't forget to remove the names from the metadata file too
```



Merge sequence tables from multiple runs

```{r, echo=FALSE}
st1 <- readRDS("~/Documents/GitHub/McH1-7_field_trials_BS2andBS3/cutadapt_NS2107_BS2/seqtab.rds")
st2 <- readRDS("~/Documents/GitHub/McH1-7_field_trials_BS2andBS3/cutadapt_NS2172_BS2/seqtab.rds") 
st3 <- readRDS("~/Documents/GitHub/McH1-7_field_trials_BS2andBS3/cutadapt_NS2211_BS2/seqtab.rds") 
st4 <- readRDS("~/Documents/GitHub/McH1-7_field_trials_BS2andBS3/cutadapt_NS2408_BS2/seqtab.rds") 
st.all <- mergeSequenceTables(st1, st2, st3, st4, repeats="sum") 

#Remove chimeric sequences:
seqtab.nochim <- removeBimeraDenovo(st.all, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
# 197 samples and 21,163 taxa
sum(seqtab.nochim)/sum(st.all)
# proportion of non-chimeric taxa: 0.9627945 (4% of ASVs were removed as chimeras)

# SAVE the non-chimeric sequence variant table SO YOU DON'T HAVE TO REPEAT ALL OF THE ABOVE STEPS
saveRDS(seqtab.nochim, file="~/Documents/GitHub/McH1-7_field_trials_BS2andBS3/probiotics.rds")

# if you need to read back in the rds file:
# seqtab.nochim <- readRDS("~/Documents/McH1-7_field_trials_BS2andBS3/probiotics.rds")
```


## Assign taxonomy in DADA2

Make sure the taxonomy reference database is in your working directory. Keep the database file gzipped. Adjust path name below. This step is very time consuming.

When taxonomy assignment is complete, we will use base R and phyloseq to clean up the taxonomy table. First, we will replace NAs and empty cells with the lowest taxonomy classification available. Second, we will use phyloseq to remove reads that are classified as Eukaryotes or unclassified at the domain level (ie, we are keeping only Bacteria and Archaea because that is what our primers target).

```{r, echo=FALSE}
taxa <- assignTaxonomy(seqtab.nochim, "~/Documents/GitHub/silva_nr99_v138.1_train_set.fa.gz", multithread=TRUE)
# FIX the NAs in the taxa table
taxon <- as.data.frame(taxa,stringsAsFactors=FALSE)
taxon$Phylum[is.na(taxon$Phylum)] <- taxon$Kingdom[is.na(taxon$Phylum)]
taxon$Class[is.na(taxon$Class)] <- taxon$Phylum[is.na(taxon$Class)]
taxon$Order[is.na(taxon$Order)] <- taxon$Class[is.na(taxon$Order)]
taxon$Family[is.na(taxon$Family)] <- taxon$Order[is.na(taxon$Family)]
taxon$Genus[is.na(taxon$Genus)] <- taxon$Family[is.na(taxon$Genus)]
write.table(taxon,"silva_taxa_table.txt",sep="\t",col.names=NA)
write.table(seqtab.nochim, "silva_otu_table.txt",sep="\t",col.names=NA)
```


Load data into phyloseq

```{r, echo=FALSE}
# Create phyloseq object from otu and taxonomy tables from dada2, along with the sample metadata.
otu <- read.table("silva_otu_table.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_taxa_table.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_BS2_tissueloss.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps 
# 21163 taxa and 196 samples 
```

Remove chloroplasts, mitochondria, and eukaryotes

```{r, echo=FALSE}
# remove chloroplasts and mitochondria and Eukaryota
get_taxa_unique(ps, "Family") #746
get_taxa_unique(ps, "Order") #428
get_taxa_unique(ps, "Kingdom") #3
ps <- subset_taxa(ps, Family !="Mitochondria")
ps <- subset_taxa(ps, Order !="Chloroplast")
ps <- subset_taxa(ps, Kingdom !="Eukaryota")
ps <- subset_taxa(ps, Kingdom !="NA")
get_taxa_unique(ps, "Family") #743
get_taxa_unique(ps, "Order") #426
get_taxa_unique(ps, "Kingdom") #2
ps
# 20164 taxa and 196 samples
```

Export tables for future reference

```{r, echo=FALSE}
# Now export cleaned otu and taxa tables from phyloseq for future reference
otu = as(otu_table(ps), "matrix")
taxon = as(tax_table(ps), "matrix")
metadata = as(sample_data(ps), "matrix")
write.table(otu,"silva_nochloronomito_otu_table.txt",sep="\t",col.names=NA)
write.table(taxon,"silva_nochloronomito_taxa_table.txt",sep="\t",col.names=NA)
```

Export ASV table with relative abundances

```{r, echo=FALSE}
# export ASV table as relative abundance
ps_ra<-transform_sample_counts(ps, function(OTU) OTU/sum(OTU))
otu_ra = as(otu_table(ps_ra), "matrix")
write.table(otu_ra,"silva_nochloronomito_otu_table_RA.txt",sep="\t",col.names=NA)
```


## Explore data from nochloronomito tables

Now, time to explore the data. First, I will now remove the blanks and save the phyloseq object and tables for the BS2 site without the blanks. Then, I will remove low abundance reads.

```{r, echo=FALSE}
# read data in, if needed
otu <- read.table("silva_nochloronomito_otu_table.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_BS2_tissueloss.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))

ps #20164 taxa and 196 samples

# Remove sample blanks
psnb = subset_samples(ps, Coral.tag != "blank")
psnb #20164 taxa and 194 samples
psnb <- prune_taxa(taxa_sums(psnb) > 1, psnb) 
psnb #20082 taxa and 194 samples, 82 taxa only in blanks

# export tables for BS2
otu_psnb = as(otu_table(psnb), "matrix")
taxon_psnb = as(tax_table(psnb), "matrix")
metadata_psnb = as(sample_data(psnb), "matrix")
write.table(otu_psnb,"silva_nochloronomito_otu_table_BS2nb.txt",sep="\t",col.names=NA)
write.table(taxon_psnb,"silva_nochloronomito_taxa_table_BS2nb.txt",sep="\t",col.names=NA)
write.table(metadata_psnb,"metadata_BS2nb.txt",sep="\t",col.names=NA)
# export relative abundances
psBS2_ra<-transform_sample_counts(psnb, function(OTU) OTU/sum(OTU))
otu_psBS2_ra = as(otu_table(psBS2_ra), "matrix")
write.table(otu_psBS2_ra,"silva_nochloronomito_otu_table_psBS2_RA.txt",sep="\t",col.names=NA)

#explore BS2 data
ntaxa(psnb) #20082
get_taxa_unique(psnb, "Order") #426
get_taxa_unique(psnb, "Class") #183

ps5_BS2<-filter_taxa(psnb, function(x) mean(x) >5, TRUE)
ntaxa(ps5_BS2) #756
get_taxa_unique(ps5_BS2, "Order") #102
get_taxa_unique(ps5_BS2, "Class") #46

# Export files with low abundance taxa removed
otu_ps5_BS2 = as(otu_table(ps5_BS2), "matrix")
taxon_ps5_BS2 = as(tax_table(ps5_BS2), "matrix")
metadata_ps5_BS2 = as(sample_data(ps5_BS2), "matrix")
write.table(otu_ps5_BS2,"silva_nochloronomito_otu_table_ps5_BS2.txt",sep="\t",col.names=NA)
write.table(taxon_ps5_BS2,"silva_nochloronomito_taxa_table_ps5_BS2.txt",sep="\t",col.names=NA)
write.table(metadata_ps5_BS2,"metadata_ps5_BS2.txt",sep="\t",col.names=NA)
# also write out ASV table with relative abundance
ps5_BS2_ra<-transform_sample_counts(ps5_BS2, function(OTU) OTU/sum(OTU))
otu_ps5_BS2_ra = as(otu_table(ps5_BS2_ra), "matrix")
write.table(otu_ps5_BS2_ra,"silva_nochloronomito_otu_table_ps5_BS2_RA.txt",sep="\t",col.names=NA)

```

## Perform center-log-ratio transformation on ASVs and calculate Aitchison Distance and principal components

## BS2

```{r, echo=FALSE}
# use an ASV table that has been filtered to remove low-abundance ASVs, no blanks
otu <- read.table("silva_nochloronomito_otu_table_ps5_BS2.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table_ps5_BS2.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_ps5_BS2.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps5_BS2 <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps5_BS2 #756 taxa and 194 samples

# First, replace 0 values with an estimate (because normalization is taking log, can't have 0)
# Also transposing here, need samples as rows
d.czm <- cmultRepl(t(otu), method="CZM", label=0, z.warning=1)
# Perform the center-log-ratio (CLR) transformation 
d.clr <- codaSeq.clr(d.czm)
# transpose matrix of CLR transformed data for ordination and dendrogram
E.clr <- t(d.clr)
# plot compositional PCA biplot (perform a singular value decomposition)
d.pcx <- prcomp(E.clr)
# calculate percent variance explained for the axis labels
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
#biplot(d.pcx, cex=c(0.6,0.4), var.axes=F,scale=1, xlab=xlab, ylab=ylab)
summary(d.pcx)
str(d.pcx)
screeplot(d.pcx)

# replot PCA with ggplot2 (showing samples only)
df_out <- as.data.frame(d.pcx$x)
theme_set(theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()))
cols<-c("resistant"="#000000","none"="#999999","probiotic paste"="#D55E00","probiotic bag"="#E69F00","control paste"="#0072B2","control bag"="#56B4E9")
samples$Treatment<-factor(samples$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples$Collection.month<-factor(samples$Collection.month,levels=c("Aug","Oct","Jan"))
samples$Condition<-factor(samples$Condition,levels=c("HH","HD","DD"))

#not using this plot for manuscript
pdf("PCA.pdf",bg ="white",width = 8.5)
p<-ggplot(df_out,aes(x=PC1,y=PC2,color=samples$Treatment,shape=samples$Condition))
p<-p+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  facet_grid(~samples$Collection.month)
p + labs(x=xlab, y=ylab, color="Treatment",shape="Condition") +coord_fixed()
dev.off()

####### Perform permanova using Aitchison distance
dist.clr <- dist(E.clr)
perm<-adonis2(dist.clr~Condition*Treatment*Collection.month,as(sample_data(ps5_BS2),"data.frame"))
print(perm)

perm2<-pairwise.adonis2(dist.clr~Condition*Treatment*Collection.month,as(sample_data(ps5_BS2),"data.frame"))
print(perm2)

perm3<-pairwise.adonis2(dist.clr~Treatment/Collection.month,as(sample_data(ps5_BS2),"data.frame"), strata='Collection.month')
print(perm3)

perm4<-pairwise.adonis2(dist.clr~Condition/Collection.month,as(sample_data(ps5_BS2),"data.frame"), strata='Collection.month')
print(perm4)

perm5<-pairwise.adonis2(dist.clr~Treatment*Condition/Collection.month,as(sample_data(ps5_BS2),"data.frame"), strata='Collection.month')
print(perm5)

#Look at diseased samples only
ps5_BS2_DD = subset_samples(ps5_BS2, Condition == "DD")
ps5_BS2_DD #755 taxa and 79 samples
otu_ps5_BS2_DD = as(otu_table(ps5_BS2_DD), "matrix")
d.czm2 <- cmultRepl(t(otu_ps5_BS2_DD), method="CZM", label=0, z.warning=1)
d.clr2 <- codaSeq.clr(d.czm2)
E.clr2 <- t(d.clr2)
dist.clr2 <- dist(E.clr2)

perm6<-pairwise.adonis2(dist.clr2~Treatment/Collection.month,as(sample_data(ps5_BS2_DD),"data.frame"), strata='Collection.month')
print(perm6)



```


Subset BS2 by month to examine treatments within time points

### August = before treatment

```{r, echo=FALSE}
# permanova within time points
ps5_BS2_aug = subset_samples(ps5_BS2, Collection.month == "Aug")
ps5_BS2_aug #755 taxa and 60 samples
otu_aug = as(otu_table(ps5_BS2_aug), "matrix")
taxon_aug = as(tax_table(ps5_BS2_aug), "matrix")
metadata_aug = as(sample_data(ps5_BS2_aug), "matrix")
write.table(otu_aug,"silva_nochloronomito_otu_table_ps5_BS2_aug.txt",sep="\t",col.names=NA)
write.table(taxon_aug,"silva_nochloronomito_taxa_table_ps5_BS2_aug.txt",sep="\t",col.names=NA)
write.table(metadata_aug,"metadata_ps5_BS2_aug.txt",sep="\t",col.names=NA)

# clear and read in again
otu <- read.table("silva_nochloronomito_otu_table_ps5_BS2_aug.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table_ps5_BS2_aug.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_ps5_BS2_aug.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps5_BS2_aug <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps5_BS2_aug  #756 taxa and 60 samples

d.czm <- cmultRepl(t(otu), method="CZM", label=0, z.warning=1)
d.clr <- codaSeq.clr(d.czm)
E.clr <- t(d.clr)
d.pcx <- prcomp(E.clr)
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
df_out.aug <- as.data.frame(d.pcx$x)
theme_set(theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()))
cols<-c("resistant"="#000000","none"="#999999","probiotic paste"="#D55E00","probiotic bag"="#E69F00","control paste"="#0072B2","control bag"="#56B4E9")
samples$Treatment<-factor(samples$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples$Condition<-factor(samples$Condition,levels=c("HH","HD","DD"))

pdf("PCA_Aug.pdf",bg ="white",width = 8.5)
p1<-ggplot(df_out.aug,aes(x=PC1,y=PC2,color=samples$Treatment,shape=samples$Condition))
p1<-p1+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  labs(x=xlab, y=ylab,color="Treatment",shape="Condition") +coord_fixed()
p1
dev.off()

# Perform permanova using Aitchison distance
dist.clr <- dist(E.clr)
perm<-adonis2(dist.clr~Condition,as(sample_data(ps5_BS2_aug),"data.frame"))
print(perm)

perm2<-pairwise.adonis2(dist.clr~Condition,as(sample_data(ps5_BS2_aug),"data.frame"))
print(perm2)

```



Subset BS2 by month to examine treatments within time points

### October = 2 weeks after treatment

```{r, echo=FALSE}
# permanova within time points
ps5_BS2_oct = subset_samples(ps5_BS2, Collection.month == "Oct")
ps5_BS2_oct #755 taxa and 71 samples
otu_oct = as(otu_table(ps5_BS2_oct), "matrix")
taxon_oct = as(tax_table(ps5_BS2_oct), "matrix")
metadata_oct = as(sample_data(ps5_BS2_oct), "matrix")
write.table(otu_oct,"silva_nochloronomito_otu_table_ps5_BS2_oct.txt",sep="\t",col.names=NA)
write.table(taxon_oct,"silva_nochloronomito_taxa_table_ps5_BS2_oct.txt",sep="\t",col.names=NA)
write.table(metadata_oct,"metadata_ps5_BS2_oct.txt",sep="\t",col.names=NA)

# clear and read in again
otu <- read.table("silva_nochloronomito_otu_table_ps5_BS2_oct.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table_ps5_BS2_oct.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_ps5_BS2_oct.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps5_BS2_oct <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps5_BS2_oct  #756 taxa and 71 samples

d.czm <- cmultRepl(t(otu), method="CZM", label=0, z.warning=1)
d.clr <- codaSeq.clr(d.czm)
E.clr <- t(d.clr)
d.pcx <- prcomp(E.clr)
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
df_out.oct <- as.data.frame(d.pcx$x)
theme_set(theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()))
cols<-c("resistant"="#000000","none"="#999999","probiotic paste"="#D55E00","probiotic bag"="#E69F00","control paste"="#0072B2","control bag"="#56B4E9")
samples$Treatment<-factor(samples$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples$Condition<-factor(samples$Condition,levels=c("HH","HD","DD"))

pdf("PCA_Oct.pdf",bg ="white",width = 8.5)
p2<-ggplot(df_out.oct,aes(x=PC1,y=PC2,color=samples$Treatment,shape=samples$Condition))
p2<-p2+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  labs(x=xlab, y=ylab, color="Treatment",shape="Condition") +coord_fixed()
p2
dev.off()

####### Perform permanova using Aitchison distance
dist.clr <- dist(E.clr)
perm<-adonis2(dist.clr~Condition*Treatment,as(sample_data(ps5_BS2_oct),"data.frame"))
print(perm)

perm2<-pairwise.adonis2(dist.clr~Condition*Treatment,as(sample_data(ps5_BS2_oct),"data.frame"))
print(perm2)

```



Subset BS2 by month to examine treatments within time points

### January = 2 months after treatment

```{r, echo=FALSE}
# permanova within time points
ps5_BS2_jan = subset_samples(ps5_BS2, Collection.month == "Jan")
ps5_BS2_jan #755 taxa and 63 samples
otu_jan = as(otu_table(ps5_BS2_jan), "matrix")
taxon_jan = as(tax_table(ps5_BS2_jan), "matrix")
metadata_jan = as(sample_data(ps5_BS2_jan), "matrix")
write.table(otu_jan,"silva_nochloronomito_otu_table_ps5_BS2_jan.txt",sep="\t",col.names=NA)
write.table(taxon_jan,"silva_nochloronomito_taxa_table_ps5_BS2_jan.txt",sep="\t",col.names=NA)
write.table(metadata_jan,"metadata_ps5_BS2_jan.txt",sep="\t",col.names=NA)

# clear and read in again
otu <- read.table("silva_nochloronomito_otu_table_ps5_BS2_jan.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table_ps5_BS2_jan.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_ps5_BS2_jan.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps5_BS2_jan <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps5_BS2_jan  #756 taxa and 63 samples

d.czm <- cmultRepl(t(otu), method="CZM", label=0, z.warning=1)
d.clr <- codaSeq.clr(d.czm)
E.clr <- t(d.clr)
d.pcx <- prcomp(E.clr)
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
df_out.jan <- as.data.frame(d.pcx$x)
theme_set(theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()))
cols<-c("resistant"="#000000","none"="#999999","probiotic paste"="#D55E00","probiotic bag"="#E69F00","control paste"="#0072B2","control bag"="#56B4E9")
samples$Treatment<-factor(samples$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples$Condition<-factor(samples$Condition,levels=c("HH","HD","DD"))

pdf("PCA_Jan.pdf",bg ="white",width = 8.5)
p3<-ggplot(df_out.jan,aes(x=PC1,y=PC2,color=samples$Treatment,shape=samples$Condition))
p3<-p3+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  labs(x=xlab, y=ylab, color="Treatment",shape="Condition") +coord_fixed()
p3
dev.off()

####### Perform PERMANOVA using Aitchison distance
dist.clr <- dist(E.clr)
perm<-adonis2(dist.clr~Condition*Treatment,as(sample_data(ps5_BS2_jan),"data.frame"))
print(perm)

perm2<-pairwise.adonis2(dist.clr~Condition*Treatment,as(sample_data(ps5_BS2_jan),"data.frame"))
print(perm2)

```



### Use cowplot to bring 3 separate PCA plots together

```{r, echo=FALSE}

#clear data and read in just what is needed for plotting

cols<-c("resistant"="#000000","none"="#999999","probiotic paste"="#D55E00","probiotic bag"="#E69F00","control paste"="#0072B2","control bag"="#56B4E9")


### AUGUST
otu.aug <- read.table("silva_nochloronomito_otu_table_ps5_BS2_aug.txt",sep="\t",header=TRUE, row.names=1)
samples.aug<-read.table("metadata_ps5_BS2_aug.txt",sep="\t",header=T,row.names=1)
samples.aug$Treatment<-factor(samples.aug$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples.aug$Condition<-factor(samples.aug$Condition,levels=c("HH","HD","DD"))
d.czm <- cmultRepl(t(otu.aug), method="CZM", label=0, z.warning=1)
d.clr <- codaSeq.clr(d.czm)
E.clr <- t(d.clr)
d.pcx <- prcomp(E.clr)
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
df_out.aug <- as.data.frame(d.pcx$x)
p1<-ggplot(df_out.aug,aes(x=PC1,y=PC2,color=samples.aug$Treatment,shape=samples.aug$Condition))
p1<-p1+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  labs(x=xlab, y=ylab, color="Treatment")+
  ylim(-100,100)+
  xlim(-100,100)+
  coord_fixed(ratio=1)

### OCTOBER
otu.oct <- read.table("silva_nochloronomito_otu_table_ps5_BS2_oct.txt",sep="\t",header=TRUE, row.names=1)
samples.oct<-read.table("metadata_ps5_BS2_oct.txt",sep="\t",header=T,row.names=1)
samples.oct$Treatment<-factor(samples.oct$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples.oct$Condition<-factor(samples.oct$Condition,levels=c("HH","HD","DD"))
d.czm <- cmultRepl(t(otu.oct), method="CZM", label=0, z.warning=1)
d.clr <- codaSeq.clr(d.czm)
E.clr <- t(d.clr)
d.pcx <- prcomp(E.clr)
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
df_out.oct <- as.data.frame(d.pcx$x)
p2<-ggplot(df_out.oct,aes(x=PC1,y=PC2,color=samples.oct$Treatment,shape=samples.oct$Condition))
p2<-p2+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  labs(x=xlab, y=ylab, color="Treatment",shape="Condition")+
  ylim(-100,100)+
  xlim(-100,100)+
  coord_fixed(ratio=1)

### JANUARY
otu.jan <- read.table("silva_nochloronomito_otu_table_ps5_BS2_jan.txt",sep="\t",header=TRUE, row.names=1)
samples.jan<-read.table("metadata_ps5_BS2_jan.txt",sep="\t",header=T,row.names=1)
samples.jan$Treatment<-factor(samples.jan$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
samples.jan$Condition<-factor(samples.jan$Condition,levels=c("HH","HD","DD"))
d.czm <- cmultRepl(t(otu.jan), method="CZM", label=0, z.warning=1)
d.clr <- codaSeq.clr(d.czm)
E.clr <- t(d.clr)
d.pcx <- prcomp(E.clr)
pc1 <- round(d.pcx$sdev[1]^2/sum(d.pcx$sdev^2),2)
pc2 <- round(d.pcx$sdev[2]^2/sum(d.pcx$sdev^2),2)
xlab <- paste("PC1: ", pc1, sep="")
ylab <- paste("PC2: ", pc2, sep="")
df_out.jan <- as.data.frame(d.pcx$x)
p3<-ggplot(df_out.jan,aes(x=PC1,y=PC2,color=samples.jan$Treatment,shape=samples.jan$Condition))
p3<-p3+geom_point(size=2)+
  theme(axis.title = element_text(size=14))+
  theme(axis.text=element_text(size=12))+
  theme(legend.title = element_text(size=14))+
  theme(legend.text = element_text(size=12))+
  theme(strip.text.x = element_text(size=12))+
  scale_color_manual(values=cols)+
  guides(fill = guide_legend(override.aes=list(shape=21)))+
  labs(x=xlab, y=ylab, color="Treatment")+
  ylim(-100,100)+
  xlim(-100,100)+
  coord_fixed(ratio=1)

#cowplot shared legends vignette: https://wilkelab.org/cowplot/articles/shared_legends.html
#get legend no longer works - use inkscape to add legend 

pdf("PCA-3panels.pdf",bg ="white",width=12,height=4.5)
panels<-plot_grid(
  p1+theme(legend.position="none"),
  p2+theme(legend.position="none"),
  p3+theme(legend.position="none"),
  align = 'vh',
  #labels=c("Before treatment","Two weeks after treatment","Two months after treatment"), #the alignment of labels is weird, add with inkscape when I add the legend
  nrow=1)
panels
dev.off()

```



Stacked bar charts - won't use for manuscript

```{r, echo=FALSE}
# load in data and create phyloseq object
otu <- read.table("silva_nochloronomito_otu_table_ps5_BS2.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table_ps5_BS2.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_ps5_BS2.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps #756 taxa and 194 samples

ps_ra<-transform_sample_counts(ps, function(OTU) OTU/sum(OTU))
ps_ra #756 taxa and 194 samples
get_taxa_unique(ps_ra, "Class") #46
get_taxa_unique(ps_ra, "Order") #102
get_taxa_unique(ps_ra, "Family") #183
get_taxa_unique(ps_ra, "Genus") #323

n <- 46
# after plotting, you can re-run the next line to create a different selection of colors
palette <- distinctColorPalette(n)

sample_data(ps_ra)$Treatment<-factor(sample_data(ps_ra)$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none","resistant"))
sample_data(ps_ra)$Collection.month<-factor(sample_data(ps_ra)$Collection.month,levels=c("Aug","Oct","Jan"))
sample_data(ps_ra)$sampled<-factor(sample_data(ps_ra)$sampled,levels=c("Before","Two weeks after","Two months after"))


pdf("barchart_Class.pdf",bg ="white",width=11)
p1=plot_bar(ps_ra ,fill="Class")+
  facet_grid(Condition~sampled,scales="free",space="free")+
  geom_bar(aes(fill=Class), stat="identity",position="stack")+
  theme_bw()+
  theme(strip.text=element_text(face="bold", size=12))+
  theme(axis.text.x=element_text(size=12))+
  theme(axis.text.y=element_text(size=12))+
  scale_fill_manual(values=palette)+
  #theme(axis.title.x = element_blank())+
  theme(legend.position = "bottom")
p1
dev.off()


#take out untreated corals
ps2 = subset_samples(ps_ra, Condition != "HH")
pdf("barchart_Class_2.pdf",bg ="white",width=11)
p1=plot_bar(ps2 ,fill="Class")+
  facet_grid(Condition~sampled,scales="free",space="free")+
  geom_bar(aes(fill=Class), stat="identity",position="stack")+
  theme_bw()+
  theme(strip.text=element_text(face="bold", size=12))+
  theme(axis.text.x=element_blank())+
  theme(axis.text.y=element_text(size=12))+
  scale_fill_manual(values=palette)+
  #theme(axis.title.x = element_blank())+
  theme(legend.position = "bottom")
p1
dev.off()


```

### Create bubble plot to show common taxa across time

Subset data for the 24 colonies that have complete or near complete sets of samples (all three time points, paired HD and DD samples)

Create bubble plots of predominant taxa

```{r, echo=FALSE}

# load in data and create phyloseq object
otu <- read.table("silva_nochloronomito_otu_table_ps5_BS2.txt",sep="\t",header=TRUE, row.names=1)
taxon <- read.table("silva_nochloronomito_taxa_table_ps5_BS2.txt",sep="\t",header=TRUE,row.names=1)
samples<-read.table("metadata_ps5_BS2.txt",sep="\t",header=T,row.names=1)
OTU = otu_table(otu, taxa_are_rows=FALSE)
taxon<-as.matrix(taxon)
TAX = tax_table(taxon)
sampledata = sample_data(samples)
ps <- phyloseq(otu_table(otu, taxa_are_rows=FALSE), 
               sample_data(samples), 
               tax_table(taxon))
ps #756 taxa and 194 samples

# get samples that have paired sample types (HD/DD) and multiple time points
ps2 = subset_samples(ps, Complete_sets == "yes")
ps2 #756 taxa and 125 samples
ps2 <- prune_taxa(taxa_sums(ps2) > 1, ps2) 
ps2 #736 taxa and 125 samples, removed 20 taxa only found in HH samples

ps_ra<-transform_sample_counts(ps2, function(OTU) OTU/sum(OTU))
ps_ra #736 taxa and 125 samples
get_taxa_unique(ps_ra, "Class") #45
get_taxa_unique(ps_ra, "Order") #96
get_taxa_unique(ps_ra, "Family") #174
get_taxa_unique(ps_ra, "Genus") #314

# Get most prevalent taxa
ps20<-prune_taxa(names(sort(taxa_sums(ps_ra),TRUE)[1:20]), ps_ra)
ps20 #20 taxa and 125 samples
get_taxa_unique(ps20, "Genus") #13

# Export phyloseq object as spreadsheets for plotting
otu_20 = as(otu_table(ps20), "matrix")
taxon_20 = as(tax_table(ps20), "matrix")
metadata_20 = as(sample_data(ps20), "matrix")
write.table(otu_20,"silva_nochloronomito_otu_table_ps20.txt",sep="\t",col.names=NA)
write.table(taxon_20,"silva_nochloronomito_taxa_table_ps20.txt",sep="\t",col.names=NA)
write.table(metadata_20,"metadata_ps20.txt",sep="\t",col.names=NA)

# read in otu table with graph-appropriate ASV names, sorted ASVs in excel by taxonomic Class (top to bottom on graph = right to left columns)
asv <- read.table("silva_nochloronomito_otu_table_ps20_v4.txt",sep="\t",header=TRUE, row.names=1)
# make sample names a column
asv <- tibble::rownames_to_column(asv, "sample")
asv_long<-melt(asv, id.vars=c("sample"))
names(asv_long)[names(asv_long)=="variable"] <- "ASV"
names(asv_long)[names(asv_long)=="value"] <- "relative_abundance"

meta<-read.table("metadata_ps20.txt",sep="\t",header=T,row.names=1)
# make sample names a column
meta <- tibble::rownames_to_column(meta, "sample")

#merge asv table and metadata
asv_meta <- merge(asv_long,meta, "sample")
write.table(asv_meta,"asv_meta.txt",sep="\t",col.names=NA)


#### read in data for plot
asv_meta<-read.table("asv_meta.txt",sep="\t",header=T,row.names=1) 

#set the plot aesthetics
cols<-c("none"="#999999","probiotic paste"="#D55E00","probiotic bag"="#E69F00","control paste"="#0072B2","control bag"="#56B4E9")
asv_meta$sampled<-factor(asv_meta$sampled,levels=c("Before treatment","Two weeks after treatment","Two months after treatment"))
asv_meta$Treatment<-factor(asv_meta$Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none"))
asv_meta$ASV<-factor(asv_meta$ASV,levels=c("Other","Roseibacillus_ASV1","Enhydrobacter_ASV1","Halomonas_ASV2","Halomonas_ASV1","Endozoicomonas_ASV3","Endozoicomonas_ASV2","Endozoicomonas_ASV1","Vibrio_ASV4","Vibrio_ASV3","Vibrio_ASV2","Vibrio_ASV1","Photobacterium_ASV1","Psychrobium_ASV1","Pseudoalteromonas_ASV2","Pseudoalteromonas_ASV1","Thalassobius_ASV1","Ruegeria_ASV1","Nautella_ASV1","Halodesulfovibrio_ASV1","Renibacterium_ASV1"))

#this factor not reflected in plot
#asv_meta$Condition<-factor(asv_meta$Condition,levels=c("HD","DD"))

#make bubble plot
bubble <- ggplot(asv_meta, aes(x=sample,y=ASV,size=relative_abundance,color=Treatment))+
  geom_point(alpha=0.8)+ # the alpha value is the opacity of the bubble
  scale_size(range = c(0,10), name="Relative abundance")+ #adjust the two numbers here to change bubble sizes
  facet_grid(. ~ sampled, scales="free")+
  theme_bw()+
  scale_color_manual(values=cols)+
  theme(axis.text.x=element_blank())+  
  theme(axis.text.y=element_text(size=10))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(legend.text = element_text(size=10))
bubble  

#I like this plot, but I just want it to arrange the samples on the x-axis by treatment type.



###use dplyr to group by sample date, then arrange by treatment using my custom orders
asv_meta2 <- asv_meta %>% group_by(sampled) %>% arrange(factor(Treatment,levels=c("probiotic bag","probiotic paste","control bag","control paste","none")), .by_group = TRUE)
# the order in the table is correct, but it doesn't change what ggplot does

bubble <- ggplot(asv_meta2, aes(x=sample,y=ASV,size=relative_abundance,color=Treatment))+
  geom_point(alpha=0.8)+ # the alpha value is the opacity of the bubble
  scale_size(range = c(0,10), name="Relative abundance")+ #adjust the two numbers here to change bubble sizes
  facet_grid(. ~ sampled, scales="free")+
  theme_bw()+
  scale_color_manual(values=cols)+
  theme(axis.text.x=element_blank())+  
  theme(axis.text.y=element_text(size=10))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(legend.text = element_text(size=10))
bubble  



###set the order of values that I want within multiple columns; remember that the y-axis has to be listed bottom to top
#myorder1<-c("Before treatment","Two weeks after treatment","Two months after treatment")
#myorder2<-c("probiotic bag","probiotic paste","control bag","control paste","none")
#myorder3<-c("Other","Roseibacillus_ASV1","Enhydrobacter_ASV1","Halomonas_ASV2","Halomonas_ASV1","Endozoicomonas_ASV3","Endozoicomonas_ASV2","Endozoicomonas_ASV1","Vibrio_ASV4","Vibrio_ASV3","Vibrio_ASV2","Vibrio_ASV1","Photobacterium_ASV1","Psychrobium_ASV1","Pseudoalteromonas_ASV2","Pseudoalteromonas_ASV1","Thalassobius_ASV1","Ruegeria_ASV1","Nautella_ASV1","Halodesulfovibrio_ASV1","Renibacterium_ASV1")




```